### PR TITLE
Fix Vault's permissions

### PR DESCRIPTION
**Pull request by flvndvd**
## Description

This PR refactors the permission system related to vaults. The aim is to keep permission logic contained within the resource itself. To achieve this, we've introduced three permission helper methods available for all resources associated with a vault:
- `canRead`: Determines if the content of a resource can be read (e.g., accessing the content of a document from within a view).
- `canList`: Determines if the contents of a resource can be listed (e.g., listing all data source views within a vault).
- `canWrite`: Determines if a resource can be modified.

We should consistently use these helper methods for managing permissions.

### CODE REVIEW

File: front/lib/resources/vault_resource.ts [hunk 1]
```diff
@@ -161,9 +161,7 @@ export class VaultResource extends BaseResource<VaultModel> {
   ): Promise<VaultResource[]> {
     const vaults = await this.baseFetch(auth);
     return vaults.filter(
-      (vault) =>
-        (auth.isAdmin() && adminsBypassACL) ||
-        auth.hasPermission([vault.acl()], "read")
+      (vault) => (auth.isAdmin() && adminsBypassACL) || vault.canRead(auth)
```
Comment (tdraier) [inline front/lib/resources/vault_resource.ts:164]: can we move the `(auth.isAdmin() && adminsBypassACL)` logic inside `canRead` ?

File: front/pages/api/w/[wId]/vaults/[vId]/data_source_views/[dsvId]/tables/[tableId]/index.ts [hunk 2]
```diff
@@ -41,13 +41,7 @@ async function handler(
...
 
   const dataSourceView = await DataSourceViewResource.fetchById(auth, dsvId);
-  // TODO(GROUPS_INFRA) Move to DataSourceViewResource.fetchById once it handles permission.
-  const hasAccessToDataSourceView =
-    auth.isAdmin() ||
-    (dataSourceView &&
-      auth.hasPermission([dataSourceView.vault.acl()], "read"));
-
-  if (!dataSourceView || !hasAccessToDataSourceView) {
+  if (!dataSourceView || !dataSourceView.canList(auth)) {
```
Comment (tdraier) [inline front/pages/api/w/[wId]/vaults/[vId]/data_source_views/[dsvId]/tables/[tableId]/index.ts:44]: Shouldn't it be `canRead` here ?

File: front/lib/resources/vault_resource.ts (hunk 1 above)
Comment (flvndvd) [reply front/lib/resources/vault_resource.ts:164]: This has been changed by @Fraggle in https://github.com/dust-tt/dust/pull/7194. Will rebase.

**Review by tdraier — APPROVED** [review]

**Review by spolu — COMMENTED** [review]
:100:

**Pull request merged**